server/events/vcs/gitlab_client.go

```diff
@@ -244,12 +244,25 @@ func (g *GitlabClient) UpdateStatus(repo models.Repo, pull models.PullRequest, s
 	case models.SuccessCommitStatus:
 		gitlabState = gitlab.Success
 	}
-	_, _, err := g.Client.Commits.SetCommitStatus(repo.FullName, pull.HeadCommit, &gitlab.SetCommitStatusOptions{
+
+	mr, err := g.GetMergeRequest(pull.BaseRepo.FullName, pull.Num)
+	if err != nil {
+		return err
+	}
+	// refTarget is set to current branch if no pipeline is assigned to the commit,
+	// otherwise it is set to the pipeline created by the merge_request_event rule
+	refTarget := pull.HeadBranch
+	if mr.Pipeline != nil {
```

Comment:
Can we combine these 2 if statements?

Reply:
How about replacing the second if with a switch case to maintain the structure in case of additional `mr.Pipeline.Source` types?

```go
	refTarget := pull.HeadBranch
	if mr.Pipeline != nil {
		switch mr.Pipeline.Source {
		case "merge_request_event":
			refTarget = fmt.Sprintf("refs/merge-requests/%d/head", pull.Num)
		}
	}
```